src/Client/Grpc/GrpcDurableTaskClient.cs

```diff
@@ -468,6 +517,64 @@ public override async Task RewindInstanceAsync(
         }
     }
 
+    /// <inheritdoc/>
+    public override async IAsyncEnumerable<HistoryEvent> StreamInstanceHistoryAsync(
+        string instanceId,
+        string? executionId = null,
+        CancellationToken cancellation = default)
+    {
+        Check.NotNullOrEmpty(instanceId);
+        Check.NotEntity(this.options.EnableEntitySupport, instanceId);
+
+        P.StreamInstanceHistoryRequest request = new()
+        {
+            InstanceId = instanceId,
+            ExecutionId = executionId,
+            ForWorkItemProcessing = false,
+        };
+
+        using AsyncServerStreamingCall<P.HistoryChunk> streamResponse =
+            this.sidecarClient.StreamInstanceHistory(request, cancellationToken: cancellation);
+
+        IAsyncStreamReader<P.HistoryChunk> responseStream = streamResponse.ResponseStream;
+
+        // Create conversion state to maintain orchestration instance across events
+        // This is required for entity-related events (EntityOperationCalled, EntityLockRequested, etc.)
+        // which need the parent orchestration instance information from ExecutionStartedEvent
+        Microsoft.DurableTask.ProtoUtils.EntityConversionState conversionState = new(insertMissingEntityUnlocks: false);
+
+        bool hasMore = true;
+        while (hasMore)
+        {
+            bool moveNextResult;
+            try
+            {
+                moveNextResult = await responseStream.MoveNext(cancellation).ConfigureAwait(false);
+            }
+            catch (RpcException e) when (e.StatusCode == StatusCode.Cancelled)
+            {
+                throw new OperationCanceledException(
+                    $"The {nameof(this.StreamInstanceHistoryAsync)} operation was canceled.", e, cancellation);
+            }
+            catch (RpcException e) when (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException($"An orchestration with the instanceId {instanceId} was not found.", e);
+            }

```

Comment:
Do we generally throw the generic exception out? Or should we be wrapping it in something?

Reply:
not applicable anymore